### ISSUE TITLE
Allow for the display of comments as a chat log rather than only on the exact move

### ISSUE THREAD

**Issue by wctgit** [SabakiHQ/Sabaki#342]
Currently you can only see comments by going to the exact move where the comment appears. This makes browsing comments very unwieldy, especially if one comment is a reply to another and so forth. Most Go UIs display comments like a chat log, with Move Number headers to indicate which move they occurred on. Usually you can click on the Move Number header to jump to that particular move to get the context of the comment. This allows one to browse the game by comments rather than strictly by moves.

The automatic move name annotation should be distinct from comments, IMO. You could have one line with the move name, followed by a multi-line text box with comments displayed sequentially in order, grouped by move number, and a scroll bar to allow reading it like a typical chat window. Personally, I would make this the default style of displaying comments. Just my opinion.

**Comment by yishn** [comment]
> Most Go UIs display comments like a chat log

I disagree, I can name a few Go editing software which do not do that, e.g. GoGui, MultiGo. Perhaps in online Go servers where there is an actual chat going on, there's a chat log. Take for example a commented game, where someone explains each move and variations in meticulous detail. You wouldn't want a chat log in this case, right? Also there's a problem how to deal with variations.

**Comment by wctgit** [comment]
Ok, I was speaking only from my limited experience. However, even if it's not as common as I thought, I still think it would be a better interface for browsing the game and being able to see the flow of comments, e.g. from KGS games.

> Take for example a commented game, where someone explains each move and variations in meticulous detail. You wouldn't want a chat log in this case, right? 

Well, first of all, if the comments were grouped by move number, and synchronized so that moving to the next move causes the scroll bar to jump to that move's first comment, then actually I don't see how it would be much of a problem, if at all. However, you can always make it a selectable choice of how to display comments, in preferences (or as a small toggle control on the comment box area). Then users can choose the method they prefer. Some games may suit one style better, other games the other style.

> Also there's a problem how to deal with variations.

I think you would just concatenate the comments (grouped by move number) from the beginning of the game to the end of the current variation, ignoring comments from other variations. If you want to see another variation's comments, just switch to that variation.

**Comment by yishn** [comment]
Interesting idea, that could indeed work...
